### PR TITLE
experimental ros2 extension & colcon v2 plugin

### PR DESCRIPTION
**Pull request by cjp256**
depends on #3197

re-PR'd from #3097 to get builds pushed to store

### CODE REVIEW

**Comment by kyrofa** [comment]
Would love to land this so folks can start using Foxy.

**Comment by cjp256** [comment]
Rebased, should be good to go.

File: snapcraft/internal/elf.py [hunk 1]
```diff
@@ -36,7 +36,7 @@
 
 
 def _ldd_resolve(soname: str, soname_path: str) -> Tuple[str, str]:
-    logger.debug(f"_ldd_resolve: {soname!r} {soname_path!r}")
+    # logger.debug(f"_ldd_resolve: {soname!r} {soname_path!r}")
```
Comment (sergiusens) [inline snapcraft/internal/elf.py:39]: just noticed this

File: snapcraft/plugins/v2/colcon.py [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+    @classmethod
+    def get_schema(cls) -> Dict[str, Any]:
+        return {
+            "$schema": "http://json-schema.org/draft-04/schema#",
+            "type": "object",
+            "additionalProperties": False,
+            "properties": {
+                "colcon-ament-cmake-args": {
+                    "type": "array",
+                    "minitems": 1,
```
Comment (sergiusens) [inline snapcraft/plugins/v2/colcon.py:81]: IIRC, long standing bug in other places, but if the statement is not minItems then it is completely ignored

Comment (sergiusens) [reply snapcraft/plugins/v2/colcon.py:81]: notice the camel casing I

Comment (cjp256) [reply snapcraft/plugins/v2/colcon.py]: hah, well i suppose minItems should be 0 then :D did that too

**Pull request merged**